ruma-events/src/lib.rs

```diff
@@ -520,6 +521,18 @@ pub struct EventDeHelper {
     pub unsigned: Option<UnsignedDeHelper>,
 }
 
+/// Helper struct to determine the msgtype from a `serde_json::value::RawValue`
+#[doc(hidden)]
+#[derive(Debug, Deserialize)]
+pub struct MessageDeHelper {
+    /// The message type field
+    pub msgtype: String,
+
+    /// Everything else in the json object
+    #[serde(flatten)]
+    pub remaining: HashMap<String, JsonValue>,
```

Comment:
```suggestion
    pub remaining: BTreeMap<String, JsonValue>,
```

We never use `HashMap` because iteration / serialization order is not deterministic there (plus perf. is worse for small amounts of entries).

Reply:
Ah yeah, I forgot about this. What about `serde_json::Map`?

Reply:
`serde_json::Map` only works if the key type is exactly `String`, for other types even if they are strings under the hood (e.g. `RoomId`) it doesn't implement any traits (or methods). So to be consistent we use `BTreeMap` everywhere (except in a single place). Also `serde_json::Map` is just a wrapper around `BTreeMap` (unless a certain feature is activated on `serde_json`).

Reply:
I guess since we're wrapping `JsonValue` it might be slightly different here. Let me think and look at the existing code again...